content/Watsonx/WatsonxAI/109.md

```diff
@@ -0,0 +1,278 @@
+---
+title: '109: Large language model application building blocks'
+timeToComplete: 45
+updated: 2024-01-30
+---
+
+<QuizAlert text='Heads Up! Quiz material will be flagged like this!' />
+
+# Large language model application building blocks
+
+This hands-on exercise will show you how to integrate large language models (LLMs) with client applications. We will review several Python code samples which can be used as “building blocks” for an LLM application.
+We will then use a simple UI prototype implemented with [Streamlit](https://streamlit.io/) to show how these building blocks can be invoked from client applications.
+
+Prompt engineering is just one of the steps in the process of integrating LLMs into business applications. It is outside the scope of this guide to provide an introduction to prompt engineering in watsonx.ai. For an introduction to prompt engineering in watsonx.ai, check out the [VEST watsonx.ai L3 Labs](/watsonx/watsonxai)
+
+## Prerequisites
```

Comment:
One addition that might be needed is a requirements.txt or some code block mentioning all dependencies to install with tested versions. For reference here's what I used:

```
load_dotenv==0.1.0
ibm_watson_machine_learning==1.0.345
streamlit==1.31.0
plotly==5.18.0
```